### PR TITLE
[FORK][FEATURE] Enable prepack algorithm for 4bit weights decompression

### PR DESCRIPTION
**Pull request by dmitry-gorokhov**

### CODE REVIEW

File: src/cpu/x64/jit_brgemm_weights_decompression_kernel.cpp [hunk 1]
```diff
@@ -61,38 +61,89 @@ void jit_brgemm_weights_decompression_kernel_t<isa>::generate() {
...
+        }
+    }
+
     size_t oc_blocks_num = div_up(jcp_.oc_size, vec_size);
     for (size_t ocb = 0; ocb < oc_blocks_num; ocb++) {
+        auto ic_iters = jcp_.decomp_buffer_dt == data_type::bf16 ? jcp_.ic_internal_size : 1;
         if (jcp_.with_scales) {
             if (jcp_.broadcast_scales) {
-                for (size_t ic = 0; ic < jcp_.ic_internal_size; ic++) {
+                for (size_t ic = 0; ic < ic_iters; ic++) {
```
Comment (xuchen-intel) [inline src/cpu/x64/jit_brgemm_weights_decompression_kernel.cpp]: The "if (jcp_.with_scales) {...}" and "if (jcp_.with_zero_points) {...}" code blocks seem duplicated. Maybe we can reuse a function.

File: src/cpu/x64/jit_brgemm_weights_decompression_kernel.cpp [hunk 2]
```diff
@@ -61,38 +61,89 @@ void jit_brgemm_weights_decompression_kernel_t<isa>::generate() {
...
-                    for (size_t ic = 0; ic < jcp_.ic_internal_size; ic++) {
-                        vpermd(vmm_scales(ocb, ic), vmm_mask(ic), vmm_tmp());
+                if (ic_iters > 1 && jcp_.decomp_buffer_dt == data_type::bf16) {
+                    uni_vmovups(vmm_tmp(0), ptr[reg_scales + ocb * vec_size * sizeof(float)]);
+                    for (size_t ic = 0; ic < ic_iters; ic++) {
+                        vpermd(vmm_scales(ocb, ic), vmm_mask(ic), vmm_tmp(0));
                     }
                 } else {
-                    uni_vmovups(vmm_scales(ocb, 0), ptr[reg_scales + ocb * vec_size * sizeof(float)]);
+                    for (size_t ic = 0; ic < ic_iters; ic++) {
```
Comment (xuchen-intel) [inline src/cpu/x64/jit_brgemm_weights_decompression_kernel.cpp]: when "if (ic_iters > 1 && jcp_.decomp_buffer_dt == data_type::bf16)" is not true. ic_iters is always 1. Should we just use the original line?

File: src/cpu/x64/jit_brgemm_weights_decompression_kernel.cpp (hunk 1 above)
Comment (dmitry-gorokhov) [reply src/cpu/x64/jit_brgemm_weights_decompression_kernel.cpp]: Thanks for the suggestion. Initialization logic is extracted into separate function

File: src/cpu/x64/jit_brgemm_weights_decompression_kernel.cpp (hunk 2 above)
Comment (dmitry-gorokhov) [reply src/cpu/x64/jit_brgemm_weights_decompression_kernel.cpp]: Agree. Removed loop

**Pull request merged**